src/commonMain/kotlin/com/github/quillraven/fleks/entity.kt

```diff
@@ -309,13 +313,15 @@ class DefaultEntityProvider(
      */
     override operator fun minusAssign(entity: Entity) {
         recycledEntities.add(entity)
-        removedEntities.set(entity.id)
+        entities.removeAt(entity.id)
     }
 
     /**
      * Returns true if and only if the given [entity] is active and part of the provider.
      */
-    override fun contains(entity: Entity): Boolean = entity.id in 0 until nextId && !removedEntities[entity.id]
+    override fun contains(entity: Entity): Boolean =
+        entity.id in 0 until nextId &&
```

Comment:
I think the first part is not necessary anymore. If the entity is not part of the `entities` bag then it is not part of the `EntityService`.